R/derive_merged.R

```diff
@@ -607,6 +612,7 @@ derive_var_merged_cat <- function(dataset,
 derive_var_merged_exist_flag <- function(dataset,
                                          dataset_add,
                                          by_vars,
+                                         keep_source_vars = exprs(),
```

Comment:
I would not add `keep_source_vars` to `derive_var_merged_exist_flag()` because the result depends on the sort order of the additional dataset. Consider for example
```
> library(dplyr, warn.conflicts = FALSE)
> 
> dm <- tribble(
+     ~STUDYID,  ~DOMAIN,  ~USUBJID, ~AGE,   ~AGEU,
+     "PILOT01",    "DM", "01-1028",   71, "YEARS",
+     "PILOT01",    "DM", "04-1127",   84, "YEARS",
+     "PILOT01",    "DM", "06-1049",   60, "YEARS"
+ )
> 
> vs <- tribble(
+     ~STUDYID,  ~DOMAIN,  ~USUBJID,      ~VISIT, ~VSTESTCD, ~VSSTRESN, ~VSBLFL,
+     "PILOT01",    "VS", "01-1028", "SCREENING",  "HEIGHT",     177.8,      NA,
+     "PILOT01",    "VS", "01-1028", "SCREENING",  "WEIGHT",     98.88,      NA,
+     "PILOT01",    "VS", "01-1028",  "BASELINE",  "WEIGHT",     99.34,     "Y",
+     "PILOT01",    "VS", "01-1028",    "WEEK 4",  "WEIGHT",     98.88,      NA,
+     "PILOT01",    "VS", "04-1127", "SCREENING",  "HEIGHT",     165.1,      NA,
+     "PILOT01",    "VS", "04-1127", "SCREENING",  "WEIGHT",     42.87,      NA,
+     "PILOT01",    "VS", "04-1127",  "BASELINE",  "WEIGHT",     41.05,     "Y",
+     "PILOT01",    "VS", "04-1127",    "WEEK 4",  "WEIGHT",     41.73,      NA,
+     "PILOT01",    "VS", "06-1049", "SCREENING",  "HEIGHT",    167.64,      NA,
+     "PILOT01",    "VS", "06-1049", "SCREENING",  "WEIGHT",     57.61,      NA,
+     "PILOT01",    "VS", "06-1049",  "BASELINE",  "WEIGHT",     57.83,     "Y",
+     "PILOT01",    "VS", "06-1049",    "WEEK 4",  "WEIGHT",     58.97,      NA
+ )
> derive_var_merged_exist_flag(
+     dm,
+     dataset_add = arrange(vs, USUBJID, desc(VISIT)),
+     by_vars = exprs(STUDYID, USUBJID),
+     filter_add = VSTESTCD == "WEIGHT",
+     new_var = WTHIFL,
+     keep_source_vars = exprs(VISIT),
+     condition = VSSTRESN > 90,
+     false_value = "N",
+     missing_value = "M"
+ ) %>%
+     select(STUDYID, USUBJID, AGE, AGEU, WTHIFL, VISIT)
# A tibble: 3 × 6
  STUDYID USUBJID   AGE AGEU  WTHIFL VISIT   
  <chr>   <chr>   <dbl> <chr> <chr>  <chr>   
1 PILOT01 01-1028    71 YEARS Y      BASELINE
2 PILOT01 04-1127    84 YEARS N      BASELINE
3 PILOT01 06-1049    60 YEARS N      BASELINE
> derive_var_merged_exist_flag(
+     dm,
+     dataset_add = arrange(vs, USUBJID, VISIT),
+     by_vars = exprs(STUDYID, USUBJID),
+     filter_add = VSTESTCD == "WEIGHT",
+     new_var = WTHIFL,
+     keep_source_vars = exprs(VISIT),
+     condition = VSSTRESN > 90,
+     false_value = "N",
+     missing_value = "M"
+ ) %>%
+     select(STUDYID, USUBJID, AGE, AGEU, WTHIFL, VISIT)
# A tibble: 3 × 6
  STUDYID USUBJID   AGE AGEU  WTHIFL VISIT 
  <chr>   <chr>   <dbl> <chr> <chr>  <chr> 
1 PILOT01 01-1028    71 YEARS Y      WEEK 4
2 PILOT01 04-1127    84 YEARS N      WEEK 4
3 PILOT01 06-1049    60 YEARS N      WEEK 4
```